### PR TITLE
Fix Terraform error caused by unavailable metrics.k8s.io/v1beta1

### PR DESCRIPTION
**Pull request by NimJay**
### Background 
* This fixed https://github.com/GoogleCloudPlatform/microservices-demo/issues/1813.
* Even though `terraform apply` would successfully deploy a functional Online Boutique, the Terraform output had errors:
```
│ Error running command 'kubectl wait --for=condition=ready pods --all -n default --timeout=-1s': exit status 1. Output: + kubectl wait
│ --for=condition=ready pods --all -n default --timeout=-1s
│ E0605 13:26:36.342439    2173 memcache.go:287] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the
│ request
```
* `kubectl wait ... pods ...` is failing because the `metrics.k8s.io/v1beta1` resource is not ready.
* See more logs in [this comment](https://github.com/GoogleCloudPlatform/microservices-demo/issues/1813#issuecomment-1576823565).
* According to [this StackOverflow post](https://stackoverflow.com/questions/63885034/gke-clusters-point-to-unavailable-apiservice-v1beta1-metrics-k8s-io) and my personal testing, this error happens on new clusters (_just_ created) — during the first few minutes.
* In the first few minutes after cluster creation, running `kubectl get apiservices` outputs:
```
NAME                               SERVICE                      AVAILABLE                  AGE
...
v1beta1.metrics.k8s.io             kube-system/metrics-server   False (MissingEndpoints)   39s
...
```
* After a few minutes, the output is:
```
NAME                               SERVICE                      AVAILABLE   AGE
...
v1beta1.metrics.k8s.io             kube-system/metrics-server   True        83m
...
```
* A solution is to wait until the `v1beta1.metrics.k8s.io` resource is ready before `kubectl wait ... pods ...`.

### Change Summary
* Before `kubectl wait`-ing for Pods, the Terraform now `kubectl wait`s for the `v1beta1.metrics.k8s.io` API service.
* We also add a timeout of 20 minutes (each) to both `kubectl wait` commands.

### Testing Procedure
* I've already tested `terraform apply` with this change, and [it worked](https://screenshot.googleplex.com/6AkrRfGHRnUYDUY.png)!
* Also, once this is merged, we can check the DeployStack build run against the `main` branch (to see if `terraform apply` works):
image

### Additional info
* Running `kubectl get apiservice v1beta1.metrics.k8s.io -o yaml` during the first few minutes after cluster creation outputs:
```
apiVersion: apiregistration.k8s.io/v1
kind: APIService
metadata:
...
  name: v1beta1.metrics.k8s.io
...
status:
  conditions:
  - lastTransitionTime: "2023-06-05T13:25:13Z"
    message: endpoints for service/metrics-server in "kube-system" have no addresses
      with port name ""
    reason: MissingEndpoints
    status: "False"
    type: Available
```

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -84,7 +84,10 @@ resource "null_resource" "apply_deployment" {
 resource "null_resource" "wait_conditions" {
   provisioner "local-exec" {
     interpreter = ["bash", "-exc"]
-    command     = "kubectl wait --for=condition=ready pods --all -n ${var.namespace} --timeout=-1s"
+    command     = <<-EOT
+    kubectl wait --for=condition=AVAILABLE apiservice/v1beta1.metrics.k8s.io --timeout=1200s
+    kubectl wait --for=condition=ready pods --all -n ${var.namespace} --timeout=1200s
+    EOT
   }
```
Comment (NimJay) [inline terraform/main.tf:91]: The use of [`<<-EOT` is documented here](https://developer.hashicorp.com/terraform/language/expressions/strings#indented-heredocs).

File: terraform/main.tf [hunk 2]
```diff
@@ -84,7 +84,10 @@ resource "null_resource" "apply_deployment" {
 resource "null_resource" "wait_conditions" {
   provisioner "local-exec" {
     interpreter = ["bash", "-exc"]
-    command     = "kubectl wait --for=condition=ready pods --all -n ${var.namespace} --timeout=-1s"
+    command     = <<-EOT
+    kubectl wait --for=condition=AVAILABLE apiservice/v1beta1.metrics.k8s.io --timeout=1200s
+    kubectl wait --for=condition=ready pods --all -n ${var.namespace} --timeout=1200s
+    EOT
```
Comment (NimJay) [inline terraform/main.tf:90]: The previously used `--timeout` of `-1s` refers to 1 week. [Related docs here](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#wait:~:text=negative%20means%20wait%20for%20a%20week.).

File: terraform/main.tf [hunk 3]
```diff
@@ -84,7 +84,10 @@ resource "null_resource" "apply_deployment" {
 resource "null_resource" "wait_conditions" {
   provisioner "local-exec" {
     interpreter = ["bash", "-exc"]
-    command     = "kubectl wait --for=condition=ready pods --all -n ${var.namespace} --timeout=-1s"
+    command     = <<-EOT
+    kubectl wait --for=condition=AVAILABLE apiservice/v1beta1.metrics.k8s.io --timeout=1200s
+    kubectl wait --for=condition=ready pods --all -n ${var.namespace} --timeout=1200s
```
Comment (bourgeoisor) [inline terraform/main.tf:89]: Is 20 minutes maybe a bit too long, or is it expected that it may take as long?

**Review by bourgeoisor — APPROVED** [review]
LGTM!

File: terraform/main.tf (hunk 3 above)
Comment (NimJay) [reply terraform/main.tf]: Good thought.
From [my local testing](https://screenshot.googleplex.com/6AkrRfGHRnUYDUY.png):
* waiting on `apiservice/v1beta1.metrics.k8s.io` took about 90s
*waiting on all Pods took about 140s
So I've doubled each value above and updated the `--timeout`s:
```
kubectl wait --for=condition=AVAILABLE apiservice/v1beta1.metrics.k8s.io --timeout=180s
kubectl wait --for=condition=ready pods --all -n ${var.namespace} --timeout=280s
```

**Pull request merged**